src/lib/datepicker/datepicker-toggle.ts

```diff
@@ -44,39 +45,43 @@ export class MatDatepickerToggle<D> implements OnChanges, OnDestroy {
   /** Whether the toggle button is disabled. */
   @Input()
   get disabled() {
-    return this._disabled === undefined ? this.datepicker.disabled : this._disabled;
+    return this._disabled === undefined ? this.datepicker.disabled : !!this._disabled;
```

Comment:
Shouldn't be `this._disabled == null`?

Reply:
Either would work

Reply:
So this `disabled` is never `null`? Because if `disabled` is `null` this comparison (`null === undefined`) won't work.

Reply:
`coerceBooleanProperty` will make sure its never null. The only reason it can be `undefined` is because we don't set an initial value for `_disabled`, so if there's no property binding for `disabled` the setter is never called and it remains `undefined`